seatunnel-translation/seatunnel-translation-base/src/main/java/org/apache/seatunnel/translation/source/CoordinatedSource.java

```diff
@@ -165,7 +165,12 @@ public void run(Collector<T> collector) throws Exception {
                                         while (flag.get()) {
                                             try {
                                                 reader.pollNext(collector);
-                                                Thread.sleep(SLEEP_TIME_INTERVAL);
+                                                if (collector.isEmptyThisPollNext()) {
+                                                    Thread.sleep(100);
+                                                } else {
+                                                    collector.resetEmptyThisPollNext();
+                                                    Thread.sleep(0L);
```

Comment:
why `Thread.sleep(0L)` ?

Reply:
After a split is processed, sleep can temporarily releases CPU resources to prevent the checkpoint flow from failing to obtain the lock. It is mentioned in this issue https://github.com/apache/seatunnel/issues/5694